pkg/kubelet/util/format/pod_test.go

```diff
@@ -117,43 +118,28 @@ func TestPods(t *testing.T) {
 	pod1 := fakeCreatePod("pod1", metav1.NamespaceDefault, "551f5a43-9f2f-11e7-a589-fa163e148d75")
 	pod2 := fakeCreatePod("pod2", metav1.NamespaceDefault, "e84a99bf-d1f9-43c2-9fa5-044ac85f794b")
 
-	testCases := []struct {
-		caseName      string
-		pods          []*v1.Pod
-		expectedValue string
-	}{
-		{"input_nil_case", nil, ""},
-		{"input_empty_case", []*v1.Pod{}, ""},
-		{"input_length_one_case", []*v1.Pod{pod1}, "pod1_default(551f5a43-9f2f-11e7-a589-fa163e148d75)"},
-		{"input_length_more_than_one_case", []*v1.Pod{pod1, pod2}, "pod1_default(551f5a43-9f2f-11e7-a589-fa163e148d75), pod2_default(e84a99bf-d1f9-43c2-9fa5-044ac85f794b)"},
-		{"input_include_nil_case", []*v1.Pod{pod1, nil}, "pod1_default(551f5a43-9f2f-11e7-a589-fa163e148d75), <nil>"},
+	pod1Obj := klog.ObjectRef{
+		Name:      "pod1",
+		Namespace: metav1.NamespaceDefault,
 	}
-
-	for _, testCase := range testCases {
-		realPods := Pods(testCase.pods)
-		assert.Equalf(t, testCase.expectedValue, realPods, "Failed to test: %s", testCase.caseName)
+	pod2Obj := klog.ObjectRef{
+		Name:      "pod2",
+		Namespace: metav1.NamespaceDefault,
 	}
-}
-
-func TestPodsWithDeletionTimestamps(t *testing.T) {
-	normalDeletionTime := metav1.Date(2017, time.September, 26, 14, 37, 50, 00, time.UTC)
-	pod1 := fakeCreatePodWithDeletionTimestamp("pod1", metav1.NamespaceDefault, "551f5a43-9f2f-11e7-a589-fa163e148d75", &normalDeletionTime)
-	pod2 := fakeCreatePodWithDeletionTimestamp("pod2", metav1.NamespaceDefault, "e84a99bf-d1f9-43c2-9fa5-044ac85f794b", &normalDeletionTime)
-
 	testCases := []struct {
 		caseName      string
 		pods          []*v1.Pod
-		expectedValue string
+		expectedValue []klog.ObjectRef
 	}{
-		{"input_nil_case", nil, ""},
-		{"input_empty_case", []*v1.Pod{}, ""},
-		{"input_length_one_case", []*v1.Pod{pod1}, "pod1_default(551f5a43-9f2f-11e7-a589-fa163e148d75):DeletionTimestamp=2017-09-26T14:37:50Z"},
-		{"input_length_more_than_one_case", []*v1.Pod{pod1, pod2}, "pod1_default(551f5a43-9f2f-11e7-a589-fa163e148d75):DeletionTimestamp=2017-09-26T14:37:50Z, pod2_default(e84a99bf-d1f9-43c2-9fa5-044ac85f794b):DeletionTimestamp=2017-09-26T14:37:50Z"},
-		{"input_include_nil_case", []*v1.Pod{pod1, nil}, "pod1_default(551f5a43-9f2f-11e7-a589-fa163e148d75):DeletionTimestamp=2017-09-26T14:37:50Z, <nil>"},
+		{"input_nil_case", nil, []klog.ObjectRef{}},
+		{"input_empty_case", []*v1.Pod{}, []klog.ObjectRef{}},
+		{"input_length_one_case", []*v1.Pod{pod1, nil}, []klog.ObjectRef{pod1Obj}},
+		{"input_length_more_than_one_case", []*v1.Pod{pod1, pod2}, []klog.ObjectRef{pod1Obj, pod2Obj}},
+		{"input_include_nil_case", []*v1.Pod{pod1, nil}, []klog.ObjectRef{pod1Obj}},
```

Comment:
Should we add a test case to reflect what these will look like when written to a log?

Reply:
Would be much easier to test in github.com/kubernetes/klog. But check the [String() ](https://github.com/kubernetes/klog/blob/14dec3377f55fd2df9cb1ed7b2abc6a5c02df42a/klog.go#L1569)method , the result shown in the log should be straightforward.